### PR TITLE
Last resort lister

### PR DESCRIPTION
**Pull request by WyriHaximus**
Works by opening a memory FD, fetching the resource ID, and interating every FD from 0 to before that resource ID and checking if php://fd/ID exists, when it does yield that ID into the list

### CODE REVIEW

File: tests/LastResortTest.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * @internal
+ */
+final class LastResortTest extends TestCase
+{
+    public function testList(): void
+    {
+        $lastResort = new LastResort();
+        $fds = iteratorOrArrayToArray($lastResort->list());
+        self::assertTrue(\count($fds) >= 3, 'Assuming to have at least 3 open file descriptors (STDIN, STDOUT, and STDERR)');
```
Comment (peter279k) [inline tests/LastResortTest.php]: Consider the following assertion:

```php
self::assertGreaterThanOrEqual(3, \count($fds), 'Assuming to have at least 3 open file descriptors (STDIN, STDOUT, and STDERR)');
```

**Comment by peter279k** [comment]
Some known issues are as follows:

- The Travis CI build is failed because the dependencies have some problems.
- The `Factory` class has the conflicts and it should be fixed.

File: src/LastResort.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<?php declare(strict_types=1);
+
+namespace WyriHaximus\FileDescriptors;
+
+final class LastResort implements ListerInterface
+{
+    public function list(): iterable
```
[the remark this answers is not in the document]
Comment (WyriHaximus) [reply src/LastResort.php]: `iterable` isn't a class or constant so there shouldn't be a performance reason to do so

File: tests/LastResortTest.php (hunk 1 above)
Comment (WyriHaximus) [reply tests/LastResortTest.php]: Good one :+1:

**Comment by WyriHaximus** [comment]
Thanks for the notes @peter279k , but I'm closing this PR (for now). As pointed out by @kelunik somewhere else the resource ID returned when casting a stream to an int doesn't map to an FD 1-on-1.

**Pull request closed without merging**